storage/paths/index.go

```diff
@@ -215,6 +532,63 @@ func (i *Index) StorageAttach(ctx context.Context, si storiface.StorageInfo, st
 
 		lastHeartbeat: time.Now(),
 	}
+
+	return nil
+}
+
+func (dbi *DBIndex) StorageDetach(ctx context.Context, id storiface.ID, url string) error {
+
+	// If url not in path urls, error out
+	// if this is only path url for this storage path, drop storage path and sector decls which have this as a storage path
+
+	var qUrls sql.NullString
+	err := dbi.harmonyDB.QueryRow(ctx, "select urls from storagelocation where storage_id=$1", string(id)).Scan(&qUrls)
```

Comment:
optional:  (since empty string is handled correctly)
qUrls := ""
err := dbi.harmonyDB.QueryRow(ctx, "select COALESCE(urls,'') from storagelocation where storage_id=$1", string(id)).Scan(&qUrls)

Reply:
done